src/pupil_labs/neon_recording/sample.py

```diff
@@ -0,0 +1,45 @@
+from typing import Literal, TypeVar
+
+import numpy as np
+import pandas as pd
+
+from pupil_labs.video.array_like import ArrayLike
+
+T = TypeVar("T", covariant=True)
+
+
+def match_ts(
```

Comment:
@dourvaris It would be nice if the matching functionality could still be used to match with external data as well. This is not easily possible when using the implementation from the `SampledStream` class. I extracted this abstract version that should be easy to use with external data too. Since all our datastreams support indexing by array, e.g. `gaze[indices]`, making use of the return value is rather simple.
Does this make sense to you?